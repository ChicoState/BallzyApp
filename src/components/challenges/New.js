import React from 'react';

import {
    View,
    Text,
    StyleSheet,
    TextInput,
    TouchableOpacity,
} from 'react-native';


class New extends React.Component {
  state = {
    title: '',
    description: '',
    price: '',
  };
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.pricetitlearea}>
        <TextInput
          style={styles.titleinput}
          placeholder='Title'
          onChangeText={(text) => {
              this.setState({
                title: text,
              });
            }}
            value={this.state.title}
        />
        <View style={{flex:.3}}/>
        <TextInput
          style={styles.priceinput}
          placeholder='Price'
          onChangeText={(text) => {
              this.setState({
                price: text,
              });
            }}
            value={this.state.price}
        />
        </View>
        <Text style={{paddingLeft: 20, paddingTop:null, fontSize: 20}}>Description:</Text>
        <TextInput
          placeholder='add description .........'
          style={styles.descriptionInput}
          numberOfLines= {8}
          maxLength= {352}
          multiline={true}
          textAlignVertical = {'top'}
          onChangeText={(text) => {
              this.setState({
                description: text,
              });
            }}
            value={this.state.description}
        />

        <View style={styles.buttonarea}>
          <TouchableOpacity
            style={styles.button}
            onPress={() => this.props.navigation.navigate('Challenges', {
              chaltitle: this.state.title,
              description: this.state.description,
              price: this.state.price,

          })}>
          <Text style={styles.buttonText}>
            Post
          </Text>
          </TouchableOpacity>
        </View>


      </View>
    );
  }
}


New.propTypes = {
  search: React.PropTypes.string,
};

var styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'stretch',
    backgroundColor: '#dcebfc'
  },
  title: {
    marginTop: 20,
    marginLeft: 20,
    fontSize: 20,
  },
  pricetitlearea: {
    padding: 5,
    height: 40,
    marginRight: 16,
    marginLeft: 16,
    marginTop: 20,
    marginBottom: 10,
    flexDirection: 'row'
  },
  buttonarea: {
    padding: 5,
    height: 40,
    marginRight: 16,
    marginLeft: 16,
    marginTop: 20,
    marginBottom: 10,
    flexDirection: 'row'
  },
  titleinput: {
    flex:7,
    backgroundColor: 'white',
    borderRadius: 5,
    height: 40,
  },
  priceinput: {
    flex:2.6,
    backgroundColor: 'white',
    borderRadius: 5,
    height: 40,
  },
  descriptionInput: {
    justifyContent:'flex-start',
    alignItems: 'flex-start',
    padding: 5,
    marginLeft: 20,
    marginRight: 20,
    borderColor: 'black',
    margin: 10,
    borderRadius: 5,
    backgroundColor: 'white'
  },
  button: {
    borderRadius: 5,
    height: 40,
    width: 90,
    borderColor: '#7F8284',
    backgroundColor: '#4392f1',
    borderRadius: 10
  },
  buttonText: {
    alignSelf: 'center',
    textAlign: 'center',
    padding: 5,
    fontSize: 20,
    color: '#FFFFFF'
  },
});

export default New;
