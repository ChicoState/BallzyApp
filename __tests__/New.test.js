'use strict';

import 'react-native';
import React from 'react';
//import React from 'react';
import New from '../src/components/challenges/New';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <New />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
