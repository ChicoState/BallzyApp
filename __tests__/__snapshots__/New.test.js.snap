exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#dcebfc",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 10,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 20,
        "padding": 5,
      }
    }>
    <TextInput
      onChangeText={[Function]}
      placeholder="Title"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "flex": 7,
          "height": 40,
        }
      }
      value="" />
    <View
      style={
        Object {
          "flex": 0.3,
        }
      } />
    <TextInput
      onChangeText={[Function]}
      placeholder="Price"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "flex": 2.6,
          "height": 40,
        }
      }
      value="" />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "paddingLeft": 20,
        "paddingTop": null,
      }
    }>
    Description:
  </Text>
  <TextInput
    maxLength={352}
    multiline={true}
    numberOfLines={8}
    onChangeText={[Function]}
    placeholder="add description ........."
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 5,
        "justifyContent": "flex-start",
        "margin": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "padding": 5,
      }
    }
    textAlignVertical="top"
    value="" />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 10,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 20,
        "padding": 5,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4392f1",
          "borderColor": "#7F8284",
          "borderRadius": 10,
          "height": 40,
          "opacity": 1,
          "width": 90,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "fontSize": 20,
            "padding": 5,
            "textAlign": "center",
          }
        }>
        Post
      </Text>
    </View>
  </View>
</View>
`;
